### PR TITLE
feat(cb2-5881): modify vehicle details header

### PR DESCRIPTION
**Pull request by leifkemp-bjss**
## Contingency Testing - Modify Vehicle Details Header

Test records now display differing headers based on vehicle type.
[CB2-5881](https://dvsa.atlassian.net/browse/CB2-5881)

## Checklist

- [ ] Branch is rebased against the latest develop/common
- [ ] Necessary `id` required prepended with `"test-"` have been checked with automation testers and added
- [ ] Code and UI has been tested manually after the additional changes
- [ ] PR title includes the JIRA ticket number
- [ ] Squashed commits contain the JIRA ticket number
- [ ] Link to the PR added to the repo
- [ ] Delete branch after merge

### CODE REVIEW

File: src/app/features/test-records/components/vehicle-header/vehicle-header.component.html [hunk 1]
```diff
@@ -66,12 +66,12 @@ <h1 class="govuk-heading-l govuk-!-margin-bottom-2">{{ test?.testTypeId || '' |
...
-          {{ (techRecord$ | async)?.vehicleConfiguration | defaultNullOrEmpty | titlecase }}
-        </span>
-        <ng-template #bodyMakeModel>
-          <span>{{ (techRecord$ | async)?.bodyMake + '-' + (techRecord$ | async)?.bodyModel | defaultNullOrEmpty }}</span>
-        </ng-template>
+        <ng-container *ngIf="techRecord$ | async as techRecord" [ngSwitch]="testResult?.vehicleType">
+          <span *ngSwitchCase="vehicleTypes.TRL"> {{ createHeader(techRecord, vehicleTypes.TRL) }} </span>
+          <span *ngSwitchCase="vehicleTypes.PSV"> {{ createHeader(techRecord, vehicleTypes.PSV) }} </span>
+          <span *ngSwitchCase="vehicleTypes.HGV"> {{ createHeader(techRecord, vehicleTypes.HGV) }}</span>
+          <span *ngSwitchDefault> Idunno </span>
```
Comment (gjulien-bjss) [inline src/app/features/test-records/components/vehicle-header/vehicle-header.component.html:73]: A better fallback message?

Comment (mcadenas-bjss) [inline src/app/features/test-records/components/vehicle-header/vehicle-header.component.html:73]: Should this be something different?

File: src/app/features/test-records/components/vehicle-header/vehicle-header.component.ts [hunk 2]
```diff
@@ -39,4 +39,24 @@ export class VehicleHeaderComponent {
   get vehicleTypes() {
     return VehicleTypes;
   }
+
+  createHeader(techRecord: TechRecordModel, vehicleType: VehicleTypes) {
+    let string = '';
+
+    switch (vehicleType) {
+      case VehicleTypes.TRL:
+        string = `${techRecord.vehicleConfiguration ? techRecord.vehicleConfiguration : ''}`;
```
Comment (gjulien-bjss) [inline src/app/features/test-records/components/vehicle-header/vehicle-header.component.ts:48]:
```suggestion
        string = techRecord.vehicleConfiguration ?? '';
```

File: src/app/features/test-records/components/vehicle-header/vehicle-header.component.html (hunk 1 above)
Comment (kpearmanbjss) [inline src/app/features/test-records/components/vehicle-header/vehicle-header.component.html:73]: If this is the fallback message when we don't know the vehicle type perhaps it should be something more meaningful?

File: src/app/features/test-records/components/vehicle-header/vehicle-header.component.ts [hunk 3]
```diff
@@ -39,4 +39,24 @@ export class VehicleHeaderComponent {
   get vehicleTypes() {
     return VehicleTypes;
   }
+
+  createHeader(techRecord: TechRecordModel, vehicleType: VehicleTypes) {
+    let string = '';
```
Comment (gjulien-bjss) [inline src/app/features/test-records/components/vehicle-header/vehicle-header.component.ts:44]:
```suggestion
```

File: src/app/features/test-records/components/vehicle-header/vehicle-header.component.html [hunk 4]
```diff
@@ -66,12 +66,12 @@ <h1 class="govuk-heading-l govuk-!-margin-bottom-2">{{ test?.testTypeId || '' |
...
-        <span *ngIf="testResult?.vehicleType === vehicleTypes.TRL; else bodyMakeModel">
-          {{ (techRecord$ | async)?.vehicleConfiguration | defaultNullOrEmpty | titlecase }}
-        </span>
-        <ng-template #bodyMakeModel>
-          <span>{{ (techRecord$ | async)?.bodyMake + '-' + (techRecord$ | async)?.bodyModel | defaultNullOrEmpty }}</span>
-        </ng-template>
+        <ng-container *ngIf="techRecord$ | async as techRecord" [ngSwitch]="testResult?.vehicleType">
+          <span *ngSwitchCase="vehicleTypes.TRL"> {{ createHeader(techRecord, vehicleTypes.TRL) }} </span>
+          <span *ngSwitchCase="vehicleTypes.PSV"> {{ createHeader(techRecord, vehicleTypes.PSV) }} </span>
+          <span *ngSwitchCase="vehicleTypes.HGV"> {{ createHeader(techRecord, vehicleTypes.HGV) }}</span>
```
Comment (andrewf-bjss) [inline src/app/features/test-records/components/vehicle-header/vehicle-header.component.html:72]: We could pass the `testResult?.vehicleType` into the function, which would remove the switch from the HTML (as we have a switch in the code already)

File: src/app/features/test-records/components/vehicle-header/vehicle-header.component.ts [hunk 5]
```diff
@@ -39,4 +39,24 @@ export class VehicleHeaderComponent {
...
+        string = techRecord.vehicleConfiguration ?? '';
+        break;
+      case VehicleTypes.PSV:
+        string = techRecord.bodyMake && techRecord.bodyModel ? `${techRecord.bodyMake}-${techRecord.bodyModel}` : '';
+        break;
+      case VehicleTypes.HGV:
+        string = techRecord.chassisMake && techRecord.chassisModel ? `${techRecord.chassisMake}-${techRecord.chassisModel}` : '';
+        break;
+      default:
+        string = '';
```
Comment (gjulien-bjss) [inline src/app/features/test-records/components/vehicle-header/vehicle-header.component.ts:57]:
```suggestion
        return ''
```

File: src/app/features/test-records/components/vehicle-header/vehicle-header.component.ts [hunk 6]
```diff
@@ -39,4 +39,24 @@ export class VehicleHeaderComponent {
   get vehicleTypes() {
     return VehicleTypes;
   }
+
+  createHeader(techRecord: TechRecordModel, vehicleType: VehicleTypes) {
+    let string = '';
+
+    switch (vehicleType) {
+      case VehicleTypes.TRL:
+        string = techRecord.vehicleConfiguration ?? '';
```
Comment (gjulien-bjss) [inline src/app/features/test-records/components/vehicle-header/vehicle-header.component.ts:48]:
```suggestion
        return techRecord.vehicleConfiguration ?? '';
```

File: src/app/features/test-records/components/vehicle-header/vehicle-header.component.ts [hunk 7]
```diff
@@ -39,4 +39,24 @@ export class VehicleHeaderComponent {
...
+        string = techRecord.bodyMake && techRecord.bodyModel ? `${techRecord.bodyMake}-${techRecord.bodyModel}` : '';
+        break;
+      case VehicleTypes.HGV:
+        string = techRecord.chassisMake && techRecord.chassisModel ? `${techRecord.chassisMake}-${techRecord.chassisModel}` : '';
+        break;
+      default:
+        string = '';
+        break;
+    }
+    return string;
```
Comment (gjulien-bjss) [inline src/app/features/test-records/components/vehicle-header/vehicle-header.component.ts:60]:
```suggestion
```

File: src/app/features/test-records/components/vehicle-header/vehicle-header.component.html (hunk 1 above)
Comment (andrewf-bjss) [inline src/app/features/test-records/components/vehicle-header/vehicle-header.component.html]: Please replace this message with something more suitable, say, the original implementation?
```suggestion
          <span *ngSwitchDefault> {{ (techRecord$ | async)?.bodyMake + '-' + (techRecord$ | async)?.bodyModel | defaultNullOrEmpty }} </span>
```

File: src/app/features/test-records/components/vehicle-header/vehicle-header.component.ts [hunk 8]
```diff
@@ -39,4 +39,24 @@ export class VehicleHeaderComponent {
   get vehicleTypes() {
     return VehicleTypes;
   }
+
+  createHeader(techRecord: TechRecordModel, vehicleType: VehicleTypes) {
```
Comment (andrewf-bjss) [inline src/app/features/test-records/components/vehicle-header/vehicle-header.component.ts:43]: We're not really creating a header here, we're getting a vehicle description - perhaps this function name better describes what it does?

```suggestion
  getVehicleDescription(techRecord: TechRecordModel, vehicleType: VehicleTypes) {
```

**Review by andrewf-bjss — CHANGES_REQUESTED** [review]

**Review by gjulien-bjss — APPROVED** [review]

**Review by andrewf-bjss — APPROVED** [review]

**Comment by andrewf-bjss** [comment]
👍🏻

**Pull request merged**